rfcs/proposed/task_group_dynamic_dependencies/task_handle_semantics.md

```diff
@@ -0,0 +1,445 @@
+
+It is proposed to add the special function to add successors to the currently executed task to handle this use-case. The API can be 
+``tbb::task_group::current_task::add_successor(task_handle& sh)`` and it has the same effect as ``make_edge`` between the ``task_handle`` or ``task_tracker`` handling
+the current task and ``sh``. 
+
+## Semantics for transferring successors from the currently executing task to the other task
+
+Lets consider the use-case where the successors of the task ``current`` are transferred to the task ``target`` owned or tracked by the ``target_handler``. 
+In this case, the API ``tbb::task_group::current_task::transfer_successors_to(target_handler)`` should be called from the body of ``current``.
+
+As it was mentioned in the parent RFC, if ``transfer_successors_to`` is called outside of task belonging to the same ``task_group``, the behavior is
+undefined.
+
+It is also useful for this API to be flexible in regard to ``target_handler`` and to allow different task states.
+
+If ``target`` task is in `created`, `scheduled` or `executing` state, this API should merge together the successors list of ``current``
+and ``target`` and sets ``target`` to have the merged successors list. It should be thread-safe to add new successors to ``current`` and ``target``
+by using the ``make_edge`` API with ``current`` or ``target`` as predecessors.
+
+If ``target`` task is in `completed` state, it does not make sense to do any merging of successors list since new dependent task for the successors
+that are transferring is already completed. In that case, the responsibility for "releasing" the successors is on the ``current`` task. The API should
+release the reference counter of all successors of ``current`` in this case.
+
+<img src="transferring_between_two_basic.png" width=800>
+
+It is clear that while transferring from ``current`` to ``target`` the successors list of ``target`` should contain both previous successors of ``target``
+and the successors of ``current``.
+
+Interesting aspect is what should be done with the successors list of ``current``.
+
+The first option is to consider ``current`` and ``target`` a separate tasks even after the transferring the successors from one to another.
+
+In this case, after the transfer, the task ``current`` will have an empty successors list, and ``target`` will have a merged successors list:
+
+<img src="transferring_between_two_separate_tracking.png" width=800>
+
+After the transfer, the successors of ``current`` and ``target`` are tracked separately and adding new successors to one of them would only
+affect the successors list of one task:
+
+<img src="transferring_between_two_separate_tracking_new_successors.png" width=800>
+
+Alternative approach is to keep tracking ``current`` and ``target`` together after transferring. This requires introducing the new state of task - a `proxy` state.
```

Comment:
I can think of examples where combining tracking is useful. I think we need a counter example, where separating is useful.  Otherwise, the choice is obvious.

Reply:
The latest version proposes the combined tracking. Added the "Eager + classic" wavefront in the advanced examples section as a motivation.